tools/tplg_parser/pga.c

```diff
@@ -124,22 +139,31 @@ int tplg_register_pga(struct tplg_context *ctx)
 	int channels = 0;
 	int ret = 0;
 
-	ret = tplg_create_pga(ctx, &volume);
+	comp = (struct sof_ipc_comp *)malloc(sizeof(*volume) + UUID_SIZE);
```

Comment:
`malloc()` returns `void *`, no need to typecast. Wouldn't it be better to assign this to `volume` here and just do `comp = &volume->comp` below